src/core/api/search.js

```diff
@@ -37,6 +41,22 @@ export function search({ api, auth = false, filters = {} }: SearchParams) {
     userAgentInfo: api.userAgentInfo,
   });
 
+  // The API says: 'The "sort" parameter "random" can only be specified when
+  // the "featured" or "recommended" parameter is also present, and the "q"
+  // parameter absent.'
+  // Let's make sure we don't send invalid filters.
+  if (newFilters.sort && newFilters.sort === SEARCH_SORT_RANDOM) {
+    if (
+      !(newFilters.recommended || newFilters.featured) ||
+      ((newFilters.recommended || newFilters.featured) && newFilters.q)
```

Comment:
If the rule states that having `sort=random` and `q` together is illegal (which is how I'm reading it), then maybe you can move this statement one line up and just make it `newFilters.q`. I.e., there doesn't seem to be a need to repeat the check for `newFilters.recommended || newFilters.featured`.

Reply:
ah, right 😅